### ISSUE TITLE
[idea] CCR dive management for a Shearwater CCR computer

### ISSUE THREAD

**Issue by Bocio** [subsurface/subsurface#498]
_Applies to Subsurface 4.6.4_

Precondition: I download dives from a Shearwater Petrel CCR controller that marks dives as "ccr dive" type. After download in Subsurface, I automagically find my dives set as "ccr".
Unfortunately Subsurface populates its UI only with diluent gas data. Oxygen and other bail-out gases added by hand are not taken in account for gas consumption. Currently Subsurface manages well only dives data coming from Poseidon rebreather units which has two pressure sensor for oxygen and diluent. IIRC this improvement applies at least to all Shearwater units which are becoming the most populars.
All rebreathers works with two gases: diluent and oxygen hence it would be advantageous adding some defaults to simplify CCR dives management.

Hereafter a complete use case for CCR handling:

1. A default cylinder (taken from preferences) is automatically populated with the correct gas got from the dive computer. The "Use" field of equipment tab is set to "CC-gas". (*)
2. A default cylinder (taken from preferences) is automatically populated with oxygen. The use field is set to "CC-gas". (**)
3. The diver compiles start/end pressure for the two cylinders.
4. Anal retentive divers would add bailout cylinders marking them as "BO-gases".
5. Info tab correctly display consumed gas for both cylinders. Right now he gets only consumed diluent. (***)
6. Graph could happily skip time gas use for ccr dives.

(*)  "Use" field drop-down would have: OC-gas | CC-gas | BO-gas | Not used (BO = bailout)

(**) In my preferences I have 3L as default cylinder so I would get the same 3L cylinder for oxygen. This would cover 99% of use cases: standard ccr units use 2x2L; 2x3L or 2x4L configuration. There is small
dir-style ccr community which uses a configuration more similar to OC: a 2x7L with diluent as back gas and a separate smaller cylinder for oxygen. In this case the diver would use D7 as default cylinder (diluent) and would simply change cylinder size for the default oxygen added by Subsurface.

(***) It's very common expressing own consumed oxygen as L/m. It's mainly a personal (metabolic) data that divers learn after hours of ccr use. It would a nice addition if Subsurface could express this value near the raw liter data in the Info tab.

Basically, point 2 and gas consumption for all gases are the new additions.
